src/piece.cpp

```diff
@@ -110,6 +110,14 @@ namespace {
       p->stepsCapture = {2 * SOUTH_WEST, 2 * SOUTH_EAST, 2 * NORTH_WEST, 2 * NORTH_EAST};
```

Comment:
(Unrelated to this line, just making a "thread")

Log showing successful movement of (existing) alfil but failure to move (new) dabbaba, as described in PR description above

```
$ ./stockfish check variants.ini 
Fairy-Stockfish 310121 by Fabian Fichter
Parsing variant: 3check-crazyhouse
Parsing variant: coffeehouse
Parsing variant: makrukhouse
Variant template 'xiangqi' does not exist.
Variant template 'janggi' does not exist.
Parsing variant: anti-losalamos
Parsing variant: indiangreat
maxRank - Invalid value 10 for type Rank
maxFile - Invalid value 10 for type File
promotionRank - Invalid value 10 for type Rank
Parsing variant: upsidedown
Parsing variant: peasant
Parsing variant: weak
Parsing variant: semitorpedo
Variant template 'janus' does not exist.
Parsing variant: pawnsonly
Parsing variant: tictactoe
Parsing variant: cfour
Parsing variant: flipersi
Parsing variant: flipello
Variant template 'grand' does not exist.
Parsing variant: shogun
Parsing variant: orda
Parsing variant: myvariant

$ ./stockfish load variants.ini 
Fairy-Stockfish 310121 by Fabian Fichter
Variant template 'xiangqi' does not exist.
Variant template 'janggi' does not exist.
Variant template 'janus' does not exist.
Variant template 'grand' does not exist.
xboard
variant myvariant
setup (-) 8x8+0_fairy rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RDBQKBAR w KQkq - 0 1
piece P& fmWfceFifmnD
piece N& N
piece B& B
piece R& R
piece Q& RB
piece A& A
piece K& KO2
piece D& D
d

 +---+---+---+---+---+---+---+---+
 | r | n | b | q | k | b | n | r |8  
 +---+---+---+---+---+---+---+---+
 | p | p | p | p | p | p | p | p |7
 +---+---+---+---+---+---+---+---+
 |   |   |   |   |   |   |   |   |6
 +---+---+---+---+---+---+---+---+
 |   |   |   |   |   |   |   |   |5
 +---+---+---+---+---+---+---+---+
 |   |   |   |   |   |   |   |   |4
 +---+---+---+---+---+---+---+---+
 |   |   |   |   |   |   |   |   |3
 +---+---+---+---+---+---+---+---+
 | P | P | P | P | P | P | P | P |2
 +---+---+---+---+---+---+---+---+
 | R | D | B | Q | K | B | A | R |1 *
 +---+---+---+---+---+---+---+---+
   a   b   c   d   e   f   g   h

Fen: rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RDBQKBAR w KQkq - 0 1
Sfen: rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RDBQKBAR b - 1
Key: 5E9769DF789A7BE3
Checkers: 
g1e3
d

 +---+---+---+---+---+---+---+---+
 | r | n | b | q | k | b | n | r |8 *
 +---+---+---+---+---+---+---+---+
 | p | p | p | p | p | p | p | p |7
 +---+---+---+---+---+---+---+---+
 |   |   |   |   |   |   |   |   |6
 +---+---+---+---+---+---+---+---+
 |   |   |   |   |   |   |   |   |5
 +---+---+---+---+---+---+---+---+
 |   |   |   |   |   |   |   |   |4
 +---+---+---+---+---+---+---+---+
 |   |   |   |   | A |   |   |   |3
 +---+---+---+---+---+---+---+---+
 | P | P | P | P | P | P | P | P |2
 +---+---+---+---+---+---+---+---+
 | R | D | B | Q | K | B |   | R |1  
 +---+---+---+---+---+---+---+---+
   a   b   c   d   e   f   g   h

Fen: rnbqkbnr/pppppppp/8/8/8/4A3/PPPPPPPP/RDBQKB1R b KQkq - 1 1
Sfen: rnbqkbnr/pppppppp/8/8/8/4A3/PPPPPPPP/RDBQKB1R w - 2
Key: 6450E4DA00EC5D64
Checkers: 
h7h6
b1b3
Error (unknown command): b1b3
d

 +---+---+---+---+---+---+---+---+
 | r | n | b | q | k | b | n | r |8  
 +---+---+---+---+---+---+---+---+
 | p | p | p | p | p | p | p |   |7
 +---+---+---+---+---+---+---+---+
 |   |   |   |   |   |   |   | p |6
 +---+---+---+---+---+---+---+---+
 |   |   |   |   |   |   |   |   |5
 +---+---+---+---+---+---+---+---+
 |   |   |   |   |   |   |   |   |4
 +---+---+---+---+---+---+---+---+
 |   |   |   |   | A |   |   |   |3
 +---+---+---+---+---+---+---+---+
 | P | P | P | P | P | P | P | P |2
 +---+---+---+---+---+---+---+---+
 | R | D | B | Q | K | B |   | R |1 *
 +---+---+---+---+---+---+---+---+
   a   b   c   d   e   f   g   h

Fen: rnbqkbnr/ppppppp1/7p/8/8/4A3/PPPPPPPP/RDBQKB1R w KQkq - 0 2
Sfen: rnbqkbnr/ppppppp1/7p/8/8/4A3/PPPPPPPP/RDBQKB1R b - 3
Key: 920736C3D788CF2B
Checkers: 
^C
```

Reply:
I also tried to move the dabbaba diagonally and only one square -- left out of the above log for simplicity.